### PR TITLE
Add xemovie provider for movies

### PR DESCRIPTION
**Pull request by JamesHawkinss**

### CODE REVIEW

File: src/providers/list/xemovie/index.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      if (script.textContent.match(/https:\/\/[a-z][0-9]\.xemovie\.com/)) {
+        // eslint-disable-next-line
+        let data = JSON.parse(JSON.stringify(eval(`(${script.textContent.replace("const data = ", "").split("};")[0]}})`)));
+        streamUrl = data.playlist[0].file;
+
+        for (const [index, subtitleTrack] of data.playlist[0].tracks.entries()) {
+          const subtitleBlob = URL.createObjectURL(
+            // eslint-disable-next-line no-await-in-loop
+            await fetch(`${CORS_PROXY_URL}${subtitleTrack.file}`).then((captionRes) => captionRes.blob())
+          ); // do this so no need for CORS errors
```
Comment (mrjvs) [inline src/providers/list/xemovie/index.ts:91]: how about just putting the proxies url in the subtitle track?

File: src/providers/list/xemovie/index.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      if (!script.textContent) continue;
+
+      if (script.textContent.match(/https:\/\/[a-z][0-9]\.xemovie\.com/)) {
+        // eslint-disable-next-line
+        let data = JSON.parse(JSON.stringify(eval(`(${script.textContent.replace("const data = ", "").split("};")[0]}})`)));
+        streamUrl = data.playlist[0].file;
+
+        for (const [index, subtitleTrack] of data.playlist[0].tracks.entries()) {
+          const subtitleBlob = URL.createObjectURL(
+            // eslint-disable-next-line no-await-in-loop
```
Comment (mrjvs) [inline src/providers/list/xemovie/index.ts:89]: just change the eslint config to disable the rule. I think its fine to have it permanently disabled

File: src/providers/list/xemovie/index.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    const url = `${CORS_PROXY_URL}https://xemovie.co/movies/${media.mediaId}/watch`;
+
+    let streamUrl = "";
+    const subtitles: MWMediaCaption[] = [];
+
+    const res = await fetch(url).then(d => d.text());
+    const scripts = Array.from(new DOMParser().parseFromString(res, "text/html").querySelectorAll("script"));
+
+    for (const script of scripts) {
+      // eslint-disable-next-line no-continue
```
Comment (mrjvs) [inline src/providers/list/xemovie/index.ts:79]: just change the eslint config to disable the rule. I think its fine to have it permanently disabled

File: src/providers/list/xemovie/index.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    const movieNodes = Array.from(movieContainer.querySelectorAll("a")).filter(link => !link.className);
+
+    const results: MWProviderMediaResult[] = movieNodes.map((node) => {
+      const parent = node.parentElement;
+      if (!parent) return;
+
+      const aElement = parent.querySelector("a");
+      if (!aElement) return;
+
+      // eslint-disable-next-line consistent-return
```
Comment (mrjvs) [inline src/providers/list/xemovie/index.ts:56]: just change the eslint config to disable the rule. I think its fine to have it permanently disabled

**Review by mrjvs — CHANGES_REQUESTED** [review]

File: src/providers/list/xemovie/index.ts (hunk 1 above)
Comment (JamesHawkinss) [reply src/providers/list/xemovie/index.ts]: this isn't possible unfortunately

Comment (lem6ns) [reply src/providers/list/xemovie/index.ts]: yep, it isn't possible due to subtitles being very special and needing to be same-origin

**Review by mrjvs — APPROVED** [review]

**Pull request merged**